keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/NotificationEventListener.kt

```diff
@@ -273,10 +273,17 @@ class NotificationEventListener(
 
         val deliveryArtifact = configAndArtifact.first.artifacts.find {
           it.reference == currentState.artifactReference
-        }?: return
+        } ?: return
 
         val currentArtifact = repository.getArtifactVersionByPromotionStatus(configAndArtifact.first, currentState.environmentName, deliveryArtifact, PromotionStatus.CURRENT)
 
+        // fetch the pinned artifact, if exists
+        val pinnedArtifact =
+          repository.getPinnedVersion(configAndArtifact.first, currentState.environmentName, deliveryArtifact.reference)?.let {
```

Comment:
Small suggestion: I find the `configAndArtifact.first` and `configAndArtifact.second` expressions confusing. You could replace that with one variable for each by doing this on line 266:
```kotlin
val (config, artifact) = currentState.artifactReference?.let {
  // ...
}
```

Reply:
ohhhh fancy! will make the change, thanks

Reply:
done